src/iteration.jl

```diff
@@ -446,15 +449,20 @@ function LazyTree(f::ROOTFile, tree::TTree, treepath, branches)
     for (b, norm_name) in res_bnames
         d[Symbol(norm_name)] = LazyBranch(f, "$treepath/$b")
     end
-    return LazyTree(NamedTuple{Tuple(keys(d))}(values(d)))
+
+    if sink == LazyTree
+	return LazyTree(NamedTuple{Tuple(keys(d))}(values(d)))
+    else
+    	return sink(Tables.CopiedColumns(d))
+    end
```

Comment:
we should probably just make this `sink(...)` and do whatever we gotta to in the constructor of `LazyTree` in the future